### ISSUE TITLE
make tarball fails when libarchive is missing xv-devel package (Trac #1977)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1969]
In testing PnF for pole deployment (using make tarball to create a deployable object):

We found 'make tarball' was failing on spts-access:
LZMA_LIBRARIES-NOTFOUND => jeb.trunk.r154481.Linux-x86_64.gcc-4.4.6/lib/tools
realpath: /scratch/tschmidt/new/icecube/icetray/work/jeb/build/LZMA_LIBRARIES-NOTFOUND
Traceback (most recent call last):
 File "/home/tschmidt/jeb/trunk/cmake/install_shlib.py", line 16, in <module>
   for f in os.listdir(srcdir):
OSError: [Errno 2] No such file or directory: ''

the libarchive.cmake tool does separately check for xv-devel (lzma) installations, and failed to find it, but it was still added to the libs to install in the tarball list, and caused the fail.  

To work around, we got xv-devel installed on spts-access.  

Likely a minor edge case, but best documented....

Migrated from https://code.icecube.wisc.edu/ticket/1977
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:14:03", 
    "description": "In testing PnF for pole deployment (using make tarball to create a deployable object):\n\nWe found 'make tarball' was failing on spts-access:\nLZMA_LIBRARIES-NOTFOUND => jeb.trunk.r154481.Linux-x86_64.gcc-4.4.6/lib/tools\nrealpath: /scratch/tschmidt/new/icecube/icetray/work/jeb/build/LZMA_LIBRARIES-NOTFOUND\nTraceback (most recent call last):\n File \"/home/tschmidt/jeb/trunk/cmake/install_shlib.py\", line 16, in <module>\n   for f in os.listdir(srcdir):\nOSError: [Errno 2] No such file or directory: ''\n\n\nthe libarchive.cmake tool does separately check for xv-devel (lzma) installations, and failed to find it, but it was still added to the libs to install in the tarball list, and caused the fail.  \n\nTo work around, we got xv-devel installed on spts-access.  \n\nLikely a minor edge case, but best documented....", 
    "reporter": "blaufuss", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067243750755", 
    "component": "cmake", 
    "summary": "make tarball fails when libarchive is missing xv-devel package", 
    "priority": "minor", 
    "keywords": "", 
    "time": "2017-03-31T14:17:02", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170622T19:43:36: blaufuss commented:

> This is still an issue when making tarballs on locations without these devel libs on cmake/trunk.
> 
> Any progress on a fix? 
> 
> Turns out, @jan is also having to work around this for pass2 tarballs in simprod1

**Comment by icecube-trac** [comment]
Trac update at 20170622T20:16:56: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "nega"

**Comment by icecube-trac** [comment]
Trac update at 20170622T20:20:17: 
* nega changed milestone from "" to "August 2017"
* nega changed priority from "normal" to "minor"

**Comment by icecube-trac** [comment]
Trac update at 20170622T22:34:38: 
* blaufuss changed resolution from "" to "fixed"
* blaufuss changed status from "assigned" to "closed"
* blaufuss commented:

> In changeset 2922:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2922"
> Handle the lzma dev and libarchive dev dependencies properly.  Presence
> of one does not promise the other.  Fixes issues when you make tarball
> 
> Closes #1969
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:33:00: nwhitehorn commented:

> In changeset 170402:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170402"
> Handle the lzma dev and libarchive dev dependencies properly.  Presence
> of one does not promise the other.  Fixes issues when you make tarball
> 
> Closes #1969
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:14:03: 
* olivas changed milestone from "August 2017" to ""
* olivas commented:

> Milestone August 2017 deleted
